tests/test_graphofgrid.cpp

```diff
@@ -0,0 +1,290 @@
+#include <config.h>
+
+#include <dune/common/version.hh>
+
+#define BOOST_TEST_MODULE GraphRepresentationOfGrid
+#define BOOST_TEST_NO_MAIN
+#include <boost/test/unit_test.hpp>
+#include <opm/grid/CpGrid.hpp>
+
+#include <opm/grid/GraphOfGrid.hpp>
+
+// basic test to check if the graph was constructed correctly
+BOOST_AUTO_TEST_CASE(SimpleGraph)
+{
+	Dune::CpGrid grid;
+	std::array<int,3> dims{2,2,2};
+	std::array<double,3> size{2.,2.,2.};
+	grid.createCartesian(dims,size);
+	Opm::GraphOfGrid gog(grid);
+
+	BOOST_REQUIRE(gog.size()==8); // number of graph vertices
+	BOOST_REQUIRE(gog.numEdges(0)==3); // each vertex has 3 neighbors
+
+	auto edgeL = gog.edgeList(2);
+	BOOST_REQUIRE(edgeL.size()==3); // neighbors of vertex 2 are: 0, 3, 6
+	BOOST_REQUIRE(edgeL[0]==1.);
+	BOOST_REQUIRE(edgeL[3]==1.);
+	BOOST_REQUIRE(edgeL[6]==1.);
+	BOOST_REQUIRE(edgeL[4]==0.); // not a neighbor (edgeL's size increased)
+
+	edgeL = gog.edgeList(10); // vertex 10 is not in the graph
+	BOOST_REQUIRE(edgeL.size()==0);
+}
+
+// test vertex contraction on a simple graph
+BOOST_AUTO_TEST_CASE(SimpleGraphWithVertexContraction)
+{
+	Dune::CpGrid grid;
+	std::array<int,3> dims{2,2,2};
+	std::array<double,3> size{2.,2.,2.};
+	grid.createCartesian(dims,size);
+	Opm::GraphOfGrid gog(grid);
+
+	auto edgeL = gog.edgeList(2);
+	gog.contractVertices(0,1);
+	BOOST_REQUIRE(gog.size()==7);
+	edgeL = gog.edgeList(0);
+	BOOST_REQUIRE(edgeL.size()==4);
+	BOOST_REQUIRE(edgeL[2]==1); // neighbor of 0
+	BOOST_REQUIRE(edgeL[3]==1); // neighbor of 1
+
+	gog.contractVertices(0,2);
+	BOOST_REQUIRE(gog.size()==6);
+	BOOST_REQUIRE(gog.getVertex(0).weight==3.);
+	edgeL = gog.edgeList(0);
+	BOOST_REQUIRE(edgeL.size()==4);
+	BOOST_REQUIRE(edgeL[3]==2);
+	BOOST_REQUIRE(gog.edgeList(3).size()==2);
+	BOOST_REQUIRE(gog.edgeList(3)[0]==2);
+	BOOST_REQUIRE_MESSAGE(gog.getVertex(1).weight==0.,"Implementation detail: \
+		Looking up vertex not present in GraphofGrid gives dummy with weight 0.");
+
+    auto v5e = gog.getVertex(5).edges;
+	BOOST_REQUIRE(v5e==gog.edgeList(5));
+	BOOST_REQUIRE(v5e==gog.edgeList(6)); // 5 and 6 have the same neighbors (1, 2 got merged)
+	BOOST_REQUIRE(v5e!=gog.edgeList(7));
+
+}
+
+#include <opm/grid/GraphOfGridWrappers.hpp>
```

Comment:
The main reason was to hide Wrappers from the first few tests. I was considering splitting the file into two after reading your first comment, but it would cut it into very small pieces. I will move it to the top.